File: oo-workshop/interface-segregation/src/main/java/com/example/mobprogramming/problem/OnlineClientImpl.java
package com.example.mobprogramming.problem;

public class OnlineClientImpl implements RestaurantInterface {
    public String name;
    public String food;
    public String price;

    public OnlineClientImpl(String name, String food, String price){
        this.name = name;
        this.food = food;
        this.price = price;
    }

    @Override
    public String acceptOnlineOrder() {
        String message = "Online order from " + this.name + " for " + this.food;
        return message;
    }

    @Override
    public String payOnline() {
        String message = "Online payment for $" + this.price + " processed";
        return message;
    }

    @Override
    public String walkInCustomerOrder() {
        //not applicable for online order
        throw new UnsupportedOperationException();
    }

    @Override
    public String payInPerson() {
        //not applicable for online order
        throw new UnsupportedOperationException();
    }
}

File: oo-workshop/tsexercises/src/interfacesegregation/onlineClientImpl.ts
import { StoreInterface } from "./storeInterface"

export class OnlineClientImpl implements StoreInterface {
    //a more complete solution may use the shopping cart from the domain modeling exercise
    public customer: string
    public product: string
    public price: string

    public constructor(name: string, food: string, price: string){
        this.customer = name
        this.product = food
        this.price = price
    }

    public acceptOnlineOrder(): string {
        const message: string = "Online order from " + this.customer + " for " + this.product
        return message
    }

    public payOnline(): string {
        const message: string = "Online payment for $" + this.price + " processed"
        return message
    }

    public walkInCustomerOrder(): string {
        //not applicable for online order
        throw new Error("not supported")
    }

    public payInPerson(): string {
        //not applicable for online order
        throw new Error("not supported")
    }
}